src/state-manager/TransactionConsensus.ts

```diff
@@ -1213,6 +1214,58 @@ class TransactionConsenus {
       }
     )
 
+    const poqoSendReceiptBinary: Route<InternalBinaryHandler<Buffer>> = {
+      name: InternalRouteEnum.binary_poqo_send_receipt,
+      handler: async (payload, respond, header) => {
+        const route = InternalRouteEnum.binary_poqo_send_receipt
+        this.profiler.scopedProfileSectionStart(route)
+        nestedCountersInstance.countEvent('internal', route)
+        profilerInstance.scopedProfileSectionStart(route, false, payload.length)
+
+        const errorHandler = (
+          errorType: RequestErrorEnum,
+          opts?: { customErrorLog?: string; customCounterSuffix?: string }
+        ): void => requestErrorHandler(route, errorType, header, opts)
+
+        
+        try {
+          const requestStream = getStreamWithTypeCheck(payload, TypeIdentifierEnum.cPoqoSendReceiptReq)
+          if (!requestStream) {
+            return errorHandler(RequestErrorEnum.InvalidRequest)
+          }
+
+          const readableReq = deserializePoqoSendReceiptReq(requestStream)
+
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(readableReq.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'binary/poqo_send_receipt: no queue entry found')
+            return
+          }
+
+          if (queueEntry.poqoReceipt) {
+            // We've already handled this
+            return
+          }
+
```

Comment:
payload verification is missing

Reply:
verification data needs to be passed and that seems like it is optional unless you can clarify more on what is required.

Reply:
Oh, okay. I thought we needed to verify the data in this handler. If it's already verified, then it's fine.